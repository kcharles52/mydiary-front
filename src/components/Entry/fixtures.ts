export const props = {
  entry_id: 1,
  title: "Title 1",
  diaryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete:jest.fn()

};
