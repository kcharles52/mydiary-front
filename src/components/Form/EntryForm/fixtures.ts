export const EntryFormProps = {
  Edit: false,
  title: "string",
  diaryTitle: "string",
  date: "string",
  diaryBody: "string",
  onSubmit: jest.fn(),
  onCancel: jest.fn(),
  onChange: jest.fn()
};

  