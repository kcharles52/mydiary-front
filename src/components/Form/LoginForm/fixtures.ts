export const LoginFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  