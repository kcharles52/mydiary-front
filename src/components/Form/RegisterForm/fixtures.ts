export const RegisterFormProps = {
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  }
  