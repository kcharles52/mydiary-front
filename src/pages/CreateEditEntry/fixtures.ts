export const props = {
  Edit: false,
  title: "Form title",
  diaryTitle: "Test Diary Title",
  date: "20-2-19",
  diaryBody: "Test Diary body",
  onSubmit: jest.fn(),
  onChange: jest.fn(),
  addEntryAction: jest.fn(),
  editEntryAction: jest.fn(),
  getEntryAction: jest.fn(() =>
    Promise.resolve({
      entry: {
        entry_id: 1,
        diaryTitle: "Title 1",
        diaryEntryBody:
          "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
        date: "12/12/2018",
        delete: jest.fn()
      }
    })
  ),
  match: { params: { entry_id: "1" } },
  editEntry: {
    entry: {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  },
  newEntry: {},
  history: { push: () => {} },
  location: { pathname: "/addEntry" }
};

export const entry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
