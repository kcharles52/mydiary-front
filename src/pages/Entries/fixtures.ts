export const props = {
  getEntriesAction: jest.fn(),
  deleteEntryAction: jest.fn(),
  entries: [
    {
      entry_id: 1,
      diaryTitle: "Title 1",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      diaryTitle: "Title 2",
      diaryEntryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  deleted: [],
  handlesDelete: jest.fn(),
  history: {push:jest.fn()},
  location: { pathname: "/home" },
  match: { url: "/home" }
};

export const fetchedEntries =[
  {
    entry_id: 1,
    diaryTitle: "Title 1",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  },
  {
    entry_id: 2,
    diaryTitle: "Title 2",
    diaryEntryBody:
      "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
    date: "12/12/2018",
    delete: jest.fn()
  }
]