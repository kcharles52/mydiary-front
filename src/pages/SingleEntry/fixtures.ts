export const props = {
  getEntryAction: (id: string) => Promise.resolve(),
  deleteEntryAction: (id: string) => Promise.resolve(),
  fetchedEntry: { entry: {} },
  match: { params: { entry_id: "1" } },
  history: { push: jest.fn() },
  location: { pathname: "/Entry/1" }
};

export const fetchedEntry = {
  entry_id: 1,
  diaryTitle: "Title 1",
  diaryEntryBody:
    "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
  date: "12/12/2018",
  delete: jest.fn()
};
