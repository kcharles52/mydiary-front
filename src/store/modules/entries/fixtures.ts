export const intialState = {
  entries: [],
  entry: {},
  new: {},
  Message: "",
  deleted: []
};

export const fetchedEntries = {
  entries: [
    {
      entry_id: 1,
      title: "Title 1",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    },
    {
      entry_id: 2,
      title: "Title 2",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
    }
  ],
  Message:"Successfully fetched entries"
};

export const editedEntry = {
    entry_id: 1,
      title: "Edited",
      diaryBody:
        "Lorem ipsum dolor sit amet consectetur, adipisicing elit. Perspiciatis commodi, hic esse necessitatibus inventore illo debitis error nesciunt officia earum. Optio totam, sapiente ",
      date: "12/12/2018",
      delete: jest.fn()
}