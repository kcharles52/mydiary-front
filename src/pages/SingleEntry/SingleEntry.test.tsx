//react libraries
import * as React from "react";

//third party libraries
import { mount, shallow } from "enzyme";

//fixtures
import { props, fetchedEntry } from "./fixtures";

import { SingleEntry, mapDispatchToProps, mapStateToProps } from ".";

describe("Render single entry", () => {
  let wrapper, shallowWrapper;

  beforeEach(() => {
    wrapper = mount(<SingleEntry {...props} />);
    shallowWrapper = shallow(<SingleEntry {...props} />);
  });

  it("should render single entry", () => {
    expect(shallowWrapper).toMatchSnapshot();
  });

  it("should delete entry on click delete", () => {
    const spy = jest.spyOn(wrapper.instance(), "deleteEntry");
    wrapper.instance().forceUpdate();
    wrapper
      .find("button")
      .last()
      .simulate("click");
    expect(spy).toBeCalled();
    spy.mockRestore();
  });

  it("should redirect to edit form on click edit", () => {
    const spy = jest.spyOn(wrapper.instance(), "editRoute");
    wrapper.instance().forceUpdate();
    wrapper
      .find("button")
      .first()
      .simulate("click");
    expect(spy).toBeCalled();
    spy.mockRestore();
  });

  describe("The mapStateToProps function", () => {
    it("should return the expected props object", () => {
      const state = {
        entries: { entry: fetchedEntry }
      };
      const props = mapStateToProps(state);
      expect(props.fetchedEntry).toEqual(state.entries.entry);
    });
  });

  describe("The mapDispatchToProps function", () => {
    let dispatch, props;
    beforeEach(() => {
      dispatch = jest.fn();
      props = mapDispatchToProps(dispatch);
    });
    it("should dispatch getEntry when getEntryAction is called", () => {
      props.getEntryAction();
      expect(dispatch).toHaveBeenCalled();
    });
    it("should dispatch getEntry when deleteEntryAction is called", () => {
      props.deleteEntryAction();
      expect(dispatch).toHaveBeenCalled();
    });
  });
});
