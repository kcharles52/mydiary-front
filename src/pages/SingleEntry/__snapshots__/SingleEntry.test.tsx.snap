// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render single entry should render single entry 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SingleEntry
    deleteEntryAction={[Function]}
    fetchedEntry={
      Object {
        "entry": Object {},
      }
    }
    getEntryAction={[Function]}
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={
      Object {
        "pathname": "/Entry/1",
      }
    }
    match={
      Object {
        "params": Object {
          "entry_id": "1",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Header />,
        <div
          className="SingleEntry"
        >
          <div
            className="EntryDetails"
          >
            <div
              className="Diaryheader"
            >
              <h1 />
              <h3 />
            </div>
            <div
              className="diarybody"
            >
              <h1 />
            </div>
            <div
              className="actionButtons"
            >
              <Button
                className="button"
                label="Edit"
                onClick={[Function]}
              />
              <Button
                className="button"
                label="Delete"
                onClick={[Function]}
              />
            </div>
          </div>
        </div>,
        <Footer />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="EntryDetails"
          >
            <div
              className="Diaryheader"
            >
              <h1 />
              <h3 />
            </div>
            <div
              className="diarybody"
            >
              <h1 />
            </div>
            <div
              className="actionButtons"
            >
              <Button
                className="button"
                label="Edit"
                onClick={[Function]}
              />
              <Button
                className="button"
                label="Delete"
                onClick={[Function]}
              />
            </div>
          </div>,
          "className": "SingleEntry",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="Diaryheader"
              >
                <h1 />
                <h3 />
              </div>,
              <div
                className="diarybody"
              >
                <h1 />
              </div>,
              <div
                className="actionButtons"
              >
                <Button
                  className="button"
                  label="Edit"
                  onClick={[Function]}
                />
                <Button
                  className="button"
                  label="Delete"
                  onClick={[Function]}
                />
              </div>,
            ],
            "className": "EntryDetails",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1 />,
                  <h3 />,
                ],
                "className": "Diaryheader",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h3",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <h1 />,
                "className": "diarybody",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h1",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <Button
                    className="button"
                    label="Edit"
                    onClick={[Function]}
                  />,
                  <Button
                    className="button"
                    label="Delete"
                    onClick={[Function]}
                  />,
                ],
                "className": "actionButtons",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "button",
                    "label": "Edit",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "className": "button",
                    "label": "Delete",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Header />,
          <div
            className="SingleEntry"
          >
            <div
              className="EntryDetails"
            >
              <div
                className="Diaryheader"
              >
                <h1 />
                <h3 />
              </div>
              <div
                className="diarybody"
              >
                <h1 />
              </div>
              <div
                className="actionButtons"
              >
                <Button
                  className="button"
                  label="Edit"
                  onClick={[Function]}
                />
                <Button
                  className="button"
                  label="Delete"
                  onClick={[Function]}
                />
              </div>
            </div>
          </div>,
          <Footer />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="EntryDetails"
            >
              <div
                className="Diaryheader"
              >
                <h1 />
                <h3 />
              </div>
              <div
                className="diarybody"
              >
                <h1 />
              </div>
              <div
                className="actionButtons"
              >
                <Button
                  className="button"
                  label="Edit"
                  onClick={[Function]}
                />
                <Button
                  className="button"
                  label="Delete"
                  onClick={[Function]}
                />
              </div>
            </div>,
            "className": "SingleEntry",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="Diaryheader"
                >
                  <h1 />
                  <h3 />
                </div>,
                <div
                  className="diarybody"
                >
                  <h1 />
                </div>,
                <div
                  className="actionButtons"
                >
                  <Button
                    className="button"
                    label="Edit"
                    onClick={[Function]}
                  />
                  <Button
                    className="button"
                    label="Delete"
                    onClick={[Function]}
                  />
                </div>,
              ],
              "className": "EntryDetails",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1 />,
                    <h3 />,
                  ],
                  "className": "Diaryheader",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": undefined,
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "h3",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <h1 />,
                  "className": "diarybody",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h1",
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <Button
                      className="button"
                      label="Edit"
                      onClick={[Function]}
                    />,
                    <Button
                      className="button"
                      label="Delete"
                      onClick={[Function]}
                    />,
                  ],
                  "className": "actionButtons",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "className": "button",
                      "label": "Edit",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "className": "button",
                      "label": "Delete",
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
